Tests/AK/TestHashMap.cpp

```diff
@@ -251,3 +251,41 @@ TEST_CASE(take)
     baz = map.take("baz"_short_string);
     EXPECT(!baz.has_value());
 }
+
+TEST_CASE(move_construct)
+{
+    HashMap<int, int> orig;
+    orig.set(1, 10);
+    orig.set(2, 20);
+    orig.set(3, 30);
+    EXPECT_EQ(orig.size(), static_cast<size_t>(3));
+    EXPECT_EQ(orig.get(2), Optional<int>(20));
+
+    HashMap<int, int> second = move(orig);
+
+    EXPECT_EQ(orig.size(), static_cast<size_t>(0));
+    EXPECT_EQ(orig.get(2), Optional<int>());
+    EXPECT_EQ(second.size(), static_cast<size_t>(3));
+    EXPECT_EQ(second.get(2), Optional<int>(20));
+}
+
+TEST_CASE(move_assign)
+{
+    HashMap<int, int> orig;
+    HashMap<int, int> second;
+    orig.set(1, 10);
+    orig.set(2, 20);
+    orig.set(3, 30);
+
+    EXPECT_EQ(orig.size(), static_cast<size_t>(3));
+    EXPECT_EQ(orig.get(2), Optional<int>(20));
+    EXPECT_EQ(second.size(), static_cast<size_t>(0));
+    EXPECT_EQ(second.get(2), Optional<int>());
+
+    second = move(orig);
+
+    EXPECT_EQ(orig.size(), static_cast<size_t>(0));
+    EXPECT_EQ(orig.get(2), Optional<int>());
+    EXPECT_EQ(second.size(), static_cast<size_t>(3));
+    EXPECT_EQ(second.get(2), Optional<int>(20));
+}
```

Comment:
To make sure these are truly moves, you can use an `AllocationGuard`, which will assert this on on-target tests

Reply:
Nope, that can't work, since the move ends up calling `HashTable::~HashTable()`, which in turn calls `free`, which does `VERIFY(s_allocation_enabled)`.

Good suggestion, but I don't see how to make it work. Also, it's not really critical anyway.